libraries/AP_HAL_Linux/Util.h

```diff
@@ -48,7 +48,7 @@ class Util : public AP_HAL::Util {
      */
     void set_system_clock(uint64_t time_utc_usec);
     const char* get_custom_log_directory() { return custom_log_directory; }
-    const char* get_custom_terrain_directory() { return custom_terrain_directory; }
+    const char* get_custom_terrain_directory() const final { return custom_terrain_directory; }
```

Comment:
Class A has method foo, Class B has method foo**o**, it isn't overriding anything.

An overriding method is implicitly virtual, it doesn't matter if you add the virtual keyword or not. Like I said above, what would final be used for, if you could make derived methods non-virtual?